airflow/contrib/operators/opsgenie_alert_operator.py

```diff
@@ -0,0 +1,139 @@
+                      with host as "https://api.opsgenie.com/"
+                      and Opsgenie API key as the connection's password
+                      (e.g. "eb243592-faa2-4ba2-a551q-1afdf565c889")
+    :type http_conn_id: str
+    :param message: The Message of the Opsgenie alert
+    :type message: str
+    :param alias: Client-defined identifier of the alert
+    :type alias: str
+    :param description: Description field of the alert
+    :type description: str
+    :param responders: Teams, users, escalations and schedules that
+                      the alert will be routed to send notifications.
+    :type responders: list[dict]
+    :param visibleTo: Teams and users that the alert will become visible
+                      to without sending any notification.
+    :type visibleTo: list[dict]
+    :param actions: Custom actions that will be available for the alert.
+    :type actions: list[str]
+    :param tags: Tags of the alert.
+    :type tags: list[str]
+    :param details: Map of key-value pairs to use as custom properties of the alert.
+    :type details: dict
+    :param entity: Entity field of the alert that is
+                    generally used to specify which domain alert is related to.
+    :type entity: str
+    :param source: Source field of the alert. Default value is
+                    IP address of the incoming request.
+    :type source: str
+    :param priority: Priority level of the alert. Default value is P3.
+    :type priority: str
+    :param user: Display name of the request owner.
+    :type user: str
+    :param note: Additional note that will be added while creating the alert.
+    :type note: str
+    :param proxy: Proxy to use to make the Opsgenie Alert API call
+    :type proxy: str
+    """
+
+    @apply_defaults
+    def __init__(self,
+                 message,
+                 http_conn_id='opsgenie_default',
+                 alias=None,
+                 description=None,
+                 responders=None,
+                 visibleTo=None,
+                 actions=None,
+                 tags=None,
+                 details=None,
+                 entity=None,
+                 source=None,
+                 priority=None,
+                 user=None,
+                 note=None,
+                 proxy=None,
+                 *args,
+                 **kwargs
+                 ):
+        super(OpsgenieAlertOperator, self).__init__(*args, **kwargs)
+
+        self.message = message
+        self.http_conn_id = http_conn_id
+        self.alias = alias
+        self.description = description
+        self.responders = responders
+        self.visibleTo = visibleTo
+        self.actions = actions
+        self.tags = tags
+        self.details = details
+        self.entity = entity
+        self.source = source
+        self.priority = priority
+        self.user = user
+        self.note = note
+        self.proxy = proxy
+        self.hook = None
+
+    def _build_opsgenie_payload(self):
```

Comment:
Shoud we move this function to Hook rather than operator?

Reply:
This is in the operator as per the suggestion from https://github.com/apache/airflow/pull/4903#discussion_r264925228. 

> ... maybe add a payload parameter to the constructor instead of copying the data? The user will provide data in the appropriate format. The operator will only perform appropriate validation.

The hook accepts a `payload` dict parameter, and the operator will do all validation and parameter parsing, which is why the function exists here.